_posts/en/newsletters/2024-02-14-newsletter.md

```diff
@@ -0,0 +1,295 @@
+
+  q4="We validate maxfeerate against the modified feerate of individual
+      package transactions, not package feerate. When can this be inaccurate?"
+  a4="When a package child transaction is rejected because its modified
+      fee rate exceeds `maxfeerate` individually, but does not if it's checked
+      as a package."
+  a4link="https://bitcoincore.reviews/28950#l-84"
+
+  q5="Given that possible inaccuracy, why not check `maxfeerate` against
+      package feerate instead?"
+  a5="Because this can cause a different inaccuracy. Suppose
+      transaction A has zero fee and B CPFPs (bumps) A. Both A and B
+      are physically large so neither exceeds `maxfeerate`. But now
+      high-fee rate C is added which spends from both A and B.
+      (This is an allowed package topology because it's only two levels,
+      although it was pointed out that the `submitpackage` RPC doesn't allow
+      this topology.) In this case, C would be accepted because much of its
+      fee is absorbed by A and B, but C should be rejected."
+  a5link="https://bitcoincore.reviews/28950#l-108"
+
+  q6="Why can’t `maxfeerate` be checked immediately after decoding like
+      `maxburnamount` is?"
+  a6="Because transaction inputs famously don't explicitly state the
+      input amount; they can only be known after looking up the parent
+      output. The fee rate requires the fee, which requires the input
+      amounts."
+  a6link="https://bitcoincore.reviews/28950#l-141"
+
+  q7="How does the `maxfeerate` check in `testmempoolaccept` RPC differ
+      from `submitpackage` RPC? Why can’t they be the same?"
+  a7="`submitpackage` uses modified fees while
+      `testmempoolaccept` uses base fees, as explained earlier.
+      Also, the fee rate check is done after the `testaccept` package
+      processing because the transactions are not added
+      to mempool and broadcasted after the processing, so we can safely
+      check `maxfeerate` and return appropriate error messages.
+      The same can't be done in `submitpackage` because the package
+      transactions might have already been accepted into the mempool and
+      broadcasted to peers, rendering the check redundant."
+  a7link="https://bitcoincore.reviews/28950#l-153"
+%}
+
+## Notable code and documentation changes
+
+_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Bitcoin Inquisition][bitcoin inquisition repo], and [BINANAs][binana
+repo]._
+
+- [Bitcoin Core #28948][] adds support for (but does not enable) [version 3 transaction
+  relay][topic v3 transaction relay], allowing any v3 transaction that
```

Comment:
Is it worth mentioning that the child (of a v3 transaction) does _not_ have to also be v3? Is that true? If the child _does_ have to also be v3, that might be worth mentioning.

Wait, what is the rule exactly? I couldn't tell from the PR description. Maybe it is:

> An unconfirmed v3 tx can have _either_ one unconfirmed parent _or_ one unconfirmed child (not both), and that parent or child can also be v3 but doesn't have to be. An unconfirmed v3 tx can also have no unconfirmed parents or children.

If that's correct, maybe consider stating that; it would have helped me understand it more easily.

Reply:
the child of a v3 tx *does* need to be v3.